clang/include/clang/3C/Constraints.h

```diff
@@ -277,9 +277,13 @@ class Constraint {
   virtual bool operator==(const Constraint &Other) const = 0;
   virtual bool operator!=(const Constraint &Other) const = 0;
   virtual bool operator<(const Constraint &Other) const = 0;
-  virtual std::string getReason() { return REASON; }
+  virtual std::string getReason() const { return REASON; }
   virtual void setReason(const std::string &Rsn) { REASON = Rsn; }
 
+  bool isUnwritable(void) const {
+    return getReason() == "Declaration in non-writable file";
```

Comment:
What about "Expression in non-writable file"?  Is it possible for the same atom to have both that and another reason for wildness?

Reply:
Even if it ins't now, it might become so later. I'll cover it anyway